Content/posts/embracing-the-ecosystem.md

```diff
@@ -0,0 +1,23 @@
+---
+date: 2024-12-08 14:00
+description: Embracing more parts of the Swift on Server ecosystem as we progress towards Vapor 5
+tags: framework, vapor5, ecosystem
+image: /static/images/posts/Vapor5.png
+author: Tim
+authorImageURL: /author-images/tim.jpg
+---
+# Embracing the Ecosystem
+
+As the march to Vapor 5 continues, we are starting to see what our packages will look like in the modern ecosystem. Part of this involves integrating the many packages that are becoming integral to Swift on the server.
+
+[JWTKit](https://github.com/vapor/jwt-kit) is the prime example of this, with the new major version taking advantage of Swift 6, [Swift Testing](https://github.com/swiftlang/swift-testing), [Benchmark](https://github.com/ordo-one/package-benchmark), and the latest improvements in [Swift Crypto](https://github.com/apple/swift-crypto). JWTKit v5 was an important release that removed a vended copy of BoringSSL to improve build times for everyone building it and was built with Swift Concurrency in mind from the start.
+
+[MultipartKit](https://github.com/vapor/multipart-kit) is the next package to get [the same treatment](https://github.com/vapor/multipart-kit/pull/100). Part of this involves migrating to Swift Testing to make it easier to write tests (it is much, much nicer!), and writing benchmark tests to ensure we make the package as performant as possible. We're also getting ready for the Swift Concurrency world by removing SwiftNIO as a dependency. This may seem like an odd step, given we're "embracing the ecosystem", but it's not needed. We can take advantage of `AsyncSequence`s to provide a full streaming API, which is helpful for large payloads. 
+
+Using Swift Concurrency means that we don't need `EventLoopFuture`s and we can make the payload type agnostic so they don't require `ByteBuffer`s. The future of Swift on the Server likely involves migrating away from `ByteBuffer`s to `AsyncSequence<ArraySlice<UInt8>>` or similar. However, the performance isn't there yet so we can allow `ByteBuffer`s to be used when desired. Removing NIO as a dependency allows packages like [Swift OpenAPI Generator](https://github.com/apple/swift-openapi-generator) to integrate MultipartKit, removing duplication across the ecosystem.
```

Comment:
This one is an extremely minor nit: `AsyncSequence<ArraySlice<UInt8>> or the like.` sounds better to my ears, though take it with a grain of salt as I'm not a native english speaker.

Reply:
`However, the performance isn't there yet so we can allow `ByteBuffer`s to be used when desired.` is decently vague that i can't figure out what you mean without digging into it.